contrib_bots/lib/spotify.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+from future import standard_library # You would need to download future.
+standard_library.install_aliases()
+import json, urllib.request, urllib.parse, urllib.error
+class SpotifyHandler(object):
+    '''
+    This bot will allow users to directly search
+    for songs, artists, and albums using the Spotify API.
+    It can be called using `@spotify song/artist/album <your_search_query>`
+    '''
+    def usage(self):
+        return '''
+        This bot will allow users to directly search
+        for songs, artists, and albums using the Spotify API.
+        It can be called using `@spotify song/artist/album <your_search_query>`
+        This will return the top 10 search results from Spotify for either
+        song/artist/album.
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+        original_content = message['content']
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'spotify':
+            return False
+        is_spotify = original_content.startswith('@spotify')
+
+        return is_spotify
+
+    def handle_message(self, message, client, state_handler):
+
+        def build_data(query_type, original_content):
```

Comment:
You're doing there `orignal_content[:5]` or  `original_content[:6]` - I guess thats because `len(query_type)`  so maybe remove content_limit and execute:
`original_content[len(query_type):]`